python/ray/_private/resource_and_label_spec.py

```diff
@@ -0,0 +1,405 @@
+
+            object_store_memory_cap = (
+                ray_constants.DEFAULT_OBJECT_STORE_MAX_MEMORY_BYTES
+            )
+
+            # Cap by shm size by default to avoid low performance, but don't
+            # go lower than REQUIRE_SHM_SIZE_THRESHOLD.
+            if sys.platform == "linux" or sys.platform == "linux2":
+                # Multiple by 0.95 to give a bit of wiggle-room.
+                # https://github.com/ray-project/ray/pull/23034/files
+                shm_avail = ray._private.utils.get_shared_memory_bytes() * 0.95
+                shm_cap = max(ray_constants.REQUIRE_SHM_SIZE_THRESHOLD, shm_avail)
+
+                object_store_memory_cap = min(object_store_memory_cap, shm_cap)
+
+            # Cap memory to avoid memory waste and perf issues on large nodes
+            if (
+                object_store_memory_cap
+                and object_store_memory > object_store_memory_cap
+            ):
+                logger.debug(
+                    "Warning: Capping object memory store to {}GB. ".format(
+                        object_store_memory_cap // 1e9
+                    )
+                    + "To increase this further, specify `object_store_memory` "
+                    "when calling ray.init() or ray start."
+                )
+                object_store_memory = object_store_memory_cap
+
+        memory = self.memory
+        if memory is None:
+            memory = avail_memory - object_store_memory
+            if memory < 100e6 and memory < 0.05 * system_memory:
+                raise ValueError(
+                    "After taking into account object store and redis memory "
+                    "usage, the amount of memory on this node available for "
+                    "tasks and actors ({} GB) is less than {}% of total. "
+                    "You can adjust these settings with "
+                    "ray.init(memory=<bytes>, "
+                    "object_store_memory=<bytes>).".format(
+                        round(memory / 1e9, 2), int(100 * (memory / system_memory))
+                    )
+                )
+
+        # Set the resolved memory and object_store_memory
+        self.object_store_memory = object_store_memory
+        self.memory = memory
+
+    @staticmethod
+    def _get_current_node_accelerator(
+        num_gpus: int, resources: Dict[str, float]
+    ) -> Tuple[AcceleratorManager, int]:
+        """
+        Returns the AcceleratorManager and accelerator count for the accelerator
+        associated with this node. This assumes each node has at most one accelerator type.
+        If no accelerators are present, returns None.
+
+        Args:
+            num_gpus: GPU count (if provided by user).
+            resources: Resource dictionary containing custom resource keys.
+
+        Returns:
+            Tuple[Optional[AcceleratorManager], int]: A tuple containing the accelerator
+            manager (or None) and the detected accelerator count (0 if none found).
```

Comment:
This function doesn't only return the number of detected accelerator but it's a merge of user configured number and auto-detected number. Could you update the comments to reflect that.